src/module.ts

```diff
@@ -1747,6 +1747,7 @@ export class BinaryModule {
 export function needsExplicitUnreachable(expr: ExpressionRef): bool {
   // not applicable if pushing a value to the stack
   switch (_BinaryenExpressionGetType(expr)) {
+    case NativeType.None:
```

Comment:
This looks like it's essentially equivalent to always returning `false`. Doesn't seem quite correct.

Reply:
So better use `break`?

Reply:
Thinking what'd be correct there. If I remember correctly, this function is necessary because Binaryen has an explicit `unreachable` type. So, for example, a block can be of type `unreachable` in Binaryen, where control flow would stop, but isn't properly terminated with an `unreachable` instruction as expected by wasm. Might well be that the first switch is unnecessary, and could be replaced with `if (_BinaryenExpressionGetType(expr)) != NativeType.Unreachable) return false;`, and only do the special checking below in order to prevent duplicate unreachables. Does that make sense? :)